### PR TITLE
US-3.3.6: Add user rank and gamification leaderboard

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
-            throw new Error ('Error when getting gamification rank')
-        }
+      } else {
+          throw new Error ('Error when getting the leaderboard')
+      }
+    })
+}
+export function getReputationStatus() {
+    const url = window.location.pathname
+    return fetch( `/rest/gamification/reputation/status?url=${url}`, {
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: yes, it is required

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js [hunk 2]
```diff
@@ -1,12 +1,24 @@
-export function getGamificationRank() {
-    return fetch( `https://www.json-generator.com/api/json/get/cqoXpWOBxK?indent=2`, {
+export function getAllUserByRank() {
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
         method: 'GET',
     }).then((resp) => {
-        if(resp && resp.ok) {
+      if(resp && resp.ok) {
             return resp.json();
-        } else {
-            throw new Error ('Error when getting gamification rank')
-        }
+      } else {
+          throw new Error ('Error when getting the leaderboard')
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 4]
```diff
@@ -177,6 +180,13 @@
           }
         )
       },
+      getUserRank() {
+          getReputationStatus().then(
+              (data) => {
+                  this.gamificationRank = data.rank
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 5]
```diff
@@ -138,6 +140,7 @@
       this.getSpacesRequestsSize();
       this.getConnectionsSize();
       this.getConnectionsRequestsSize();
+      this.getUserRank();
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/GamificationRank.vue [hunk 6]
```diff
@@ -150,29 +147,50 @@
...
+        leaderBoardArray: [],
+        listBelowPoduim: []
+      }
+    },
+    created() {
+      this.getAllUserByRank();
+    },
+
+    methods: {
+      getAllUserByRank() {
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/GamificationRank.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/GamificationRank.vue [hunk 7]
```diff
@@ -150,29 +147,50 @@
...
+        getAllUserByRank().then(
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/GamificationRank.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/GamificationRank.vue [hunk 8]
```diff
@@ -150,29 +147,50 @@
...
+  import {getAllUserByRank} from "../profilStatsAPI";
+  export default {
+    data() {
+      return {
+        leaderBoardArray: [],
+        listBelowPoduim: []
+      }
+    },
+    created() {
+      this.getAllUserByRank();
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/GamificationRank.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/GamificationRank.vue [hunk 9]
```diff
@@ -150,29 +147,50 @@
...
 
 <script>
-    import * as profileStatsAPI from '../profilStatsAPI'
-    export default {
-        name: "GamificationRank",
-      data() {
-        return {
-          leaderBoardArray: []
-        }
+  import {getAllUserByRank} from "../profilStatsAPI";
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/GamificationRank.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (azayati) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: Spec should be updated

**Comment by azayati** [comment]
Related [gamifications PR](https://github.com/exo-addons/gamification/pull/84)

**Pull request merged**